lib/user/user-client.js

```diff
@@ -29,13 +31,65 @@ var xhr = function(_options) {
   return deferred.promise;
 };
 
-var storeProfile = function(profileData) {
-  localStorage.setItem('fh.wfm.profileData', JSON.stringify(profileData));
+/**
+* Encrypt the users profile data using AES with a hash of the session key and store the result in localStorage.
+* @param profileData {object} - the plaintext to encrypt.
+* @param sessionToken {string} - the secret key to encrypt with.
+*/
+var storeProfile = function(profileData, sessionToken) {
+  if (sessionToken && profileData) {
+
+    var hashedSessionKey = sha256.hash(sessionToken);
+    // handle failed hashing of session token
+    if (hashedSessionKey instanceof Error) {
+      storeLocalStorage('fh.wfm.profileData', null, hashedSessionKey.toString());
+      return;
+    }
+
+    var encryptedData = aes.encrypt(profileData, hashedSessionKey);
+    // handle failed encryption of profile data
+    if (encryptedData instanceof Error) {
+      storeLocalStorage('fh.wfm.profileData', null, encryptedData.toString());
+      return;
+    }
+
+    // set the encrypted data in localStorage
+    storeLocalStorage('fh.wfm.profileData', encryptedData, null);
+
+  } else {
+    storeLocalStorage('fh.wfm.profileData', null, "Session Token or Profile Data is Null. Setting profile data to null.");
+  }
+};
+
+/**
+* Helper method to store data in localstorage and log an error if one is provided.
+* @param key {string} - the localstorage key name.
+* @param data {object/string} - the data to store.
+* @param errMsg {string} - the error message to log.
+*/
+var storeLocalStorage = function(key, data, errMsg) {
+  localStorage.setItem(key, data);
+  if (errMsg) {
+    console.log(errMsg, "Setting profile data to null.");
```

Comment:
If no session token exists, or no profile data is returned from authResponse, it should be set to null, like before.